packages/dashboard/src/components/DashboardGridLayout/DashboardGridLayoutWithAddCardPanel/DashboardGridLayoutWithAddCardPanel.tsx

```diff
@@ -0,0 +1,200 @@
+import * as React from 'react';
+import { Layout } from 'react-grid-layout';
+import {
+  IDashboardGridLayoutWithAddCardPanelProps,
+  IDashboardGridLayoutWithAddCardPanelState
+} from './DashboardGridLayoutWithAddCardPanel.types';
+import {
+  AddCardPanel,
+  DashboardGridSectionLayout,
+  IDGLCard,
+  IDashboardCardLayout,
+  ISection,
+  CardSize,
+  DashboardGridBreakpointLayouts
+} from '../../../index';
+import { getCardStyles, getClassNames } from './DashboardGridLayoutWithAddCardPanel.styles';
+
+export class DashboardGridLayoutWithAddCardPanel extends React.Component<
+  IDashboardGridLayoutWithAddCardPanelProps,
+  IDashboardGridLayoutWithAddCardPanelState
+> {
+  private _cardsForLayout: IDGLCard[] = [];
+  private _cardsForAddCardPanel: IDGLCard[] = [];
+
+  constructor(props: IDashboardGridLayoutWithAddCardPanelProps) {
+    super(props);
+    this.state = {
+      cardsForAddCardPanel: [],
+      cardsForLayout: [],
+      sections: [],
+      layout: {
+        lg: [{ i: 'section0', y: 0, x: 0, size: CardSize.section }]
+      }
+    };
+  }
+
+  public componentDidUpdate(): void {
+    if (this._cardsForAddCardPanel !== this.props.cardsVisibleInAddCardPanel || this._cardsForLayout !== this.props.cardsVisibleInLayout) {
+      this._cardsForAddCardPanel = this.props.cardsVisibleInAddCardPanel;
+      this._cardsForLayout = this.props.cardsVisibleInLayout;
```

Comment:
>cardsVisibleInLayout [](start = 40, length = 20)

dashboardCards should be enough we don't need to say visible in dashboard. Similarly addCardPanelCards should be enough